### ISSUE TITLE
Replication stopping after having indexed a quarter of the documents

### ISSUE THREAD

**Issue by obierlaire** [couchbaselabs/elasticsearch-transport-couchbase#54]
Hi all, 
I've the following configuration:
- couchbase 2.5 with more than 2 million docs - 3 nodes
- elasticsearch 1.3 - 3 nodes
- elasticsearch-transport-couchbase 2.0

I'm running a XDCR replication and that is indexing fast but I'm having only a quarter of documents in my Elasticsearch index.

Elasticsearch does not show any particular logs except known malformed document issues (couple of hundred, defintely not explaining the 1.5 million missing). 
Couchbase however shows a large number of error logs. A few hundred  but around 500 000 doc indexed, it keeps logging a huge number of error logs per seconds over and over.

In the XDCR console:
```
2014-12-02 18:41:54 [Vb Rep] Error replicating vbucket 630. Please see logs for details.
2014-12-02 18:41:54 [Vb Rep] Error replicating vbucket 612. Please see logs for details.
2014-12-02 18:41:54 [Vb Rep] Error replicating vbucket 564. Please see logs for details.
...
```

In XDCR_errors logs we have two kind of logs going over and over:
```
[xdcr:error,2014-12-02T19:03:04.091,ns_1@10.100.174.119:<0.14777.1543>:xdc_vbucket_rep:init_replication_state:571]WARNING! Database delete purger current sequence is ahead of replicator starting sequence for (source:<<"soter">>, target:<<"/remoteClusters/cfb93ac0e62932d4983cc0f0f437cdef/buckets/soter18">>) that means one or more deletion is lost (vb: 697, purger seq: 2754, repl start seq: 0).
[xdcr:error,2014-12-02T19:03:04.135,ns_1@10.100.174.119:<0.14801.1543>:xdc_vbucket_rep:init_replication_state:571]WARNING! Database delete purger current sequence is ahead of replicator starting sequence for (source:<<"soter">>, target:<<"/remoteClusters/cfb93ac0e62932d4983cc0f0f437cdef/buckets/soter18">>) that means one or more deletion is lost (vb: 779, purger seq: 2934, repl start seq: 0).
```
Then
```
[xdcr:error,2014-12-02T19:03:04.476,ns_1@10.100.174.119:<0.12909.1543>:xdc_vbucket_rep:terminate:381]Replication (CAPI mode) `cfb93ac0e62932d4983cc0f0f437cdef/soter/soter18` (`soter/683` -> `http://*****@10.100.142.201:8070/soter18%2f683%3beec8a424a5394578b3bc7ac369f96d68`) failed: {http_request_failed,"POST",
                     "http://Administrator:*****@10.100.142.201:8070/soter18%2f683%3beec8a424a5394578b3bc7ac369f96d68/_bulk_docs",
                     {error,{code,503}}}
[xdcr:error,2014-12-02T19:03:04.557,ns_1@10.100.174.119:<0.12891.1543>:xdc_vbucket_rep:terminate:381]Replication (CAPI mode) `cfb93ac0e62932d4983cc0f0f437cdef/soter/soter18` (`soter/772` -> `http://*****@10.100.142.200:8070/soter18%2f772%3beec8a424a5394578b3bc7ac369f96d68`) failed: {http_request_failed,"POST",
                     "http://Administrator:*****@10.100.142.200:8070/soter18%2f772%3beec8a424a5394578b3bc7ac369f96d68/_revs_diff",
                     {error,{code,503}}}
```

I don't see anything in the Elasticsearch logs at the same time.

Those error logs are generated over and over and does not even seem to ever stop... It sounds like ES (and maybe CAPI or transport plugin) just refused to continue accepting XDCR replications.

Also, if I delete the xdcr replication, create a new index in ES and then create a new replication from the same bucket into this new index in ES, it just does not start. A couple of couchbaseCheckpoint is created and no document is coming. However if I restart elasticsearch before creating the replication, it all start nicely (and eventually stop after 600 000 documents)

Is this a bug or this can be fixed by tuning some settings/configuration?

**Comment by ltamrazov** [comment]
Hi, 

We are having a similar issue. We start up replication, (which just a few documents). Everything gets replicated correctly and is indexed nicely. Then at some point (usually within half hour or so) we get replication errors (even though no new documents have been added) and from that point onward, if we try to add new documents, they do not get replicated. 

At first the replication is successful, and all the documents get replicated ( so it is not the case with invalid JSON). But after a while it crashes without us having made any changes to the dataset. 

I've checked that couchbaseCheckpoint documents are present, and each contains a source with uuid. The structure of each is as follows:
```
    {
        _index: main_v1
        _type: couchbaseCheckpoint
        _id: vbucket3UUID 
        _version: 1
        _score: 1
        _source: {
            doc: {
                   uuid: 49525f6d9e6e401a968f696aa1c9f974
            }
        }
    }
```
Elastic logs are clean. Couchbase logs keep printing the following:

couchdb.log and xdcr_errors.log both print this:

```
[couchdb:info,2015-02-12T19:24:57.385,ns_1@couchbase.dev.bestforme.com:   <0.10441.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f18/_bulk_docs in 0.5 seconds due to error {code,503}
[couchdb:info,2015-02-12T19:24:57.512,ns_1@couchbase.dev.bestforme.com:<0.10459.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f60/_revs_diff in 0.5 seconds due to error {code,503}
[couchdb:error,2015-02-12T19:24:57.890,ns_1@couchbase.dev.bestforme.com:<0.10441.10>:couch_log:error:44]Replicator, request POST to "http://Administrator:*****@192.168.1.100:9091/main_v1%2f18/_bulk_docs" failed due to error {code,503}
[couchdb:info,2015-02-12T19:24:57.901,ns_1@couchbase.dev.bestforme.com:<0.10479.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f18/_bulk_docs in 0.25 seconds due to error {code,503}
[couchdb:error,2015-02-12T19:24:58.018,ns_1@couchbase.dev.bestforme.com:<0.10459.10>:couch_log:error:44]Replicator, request POST to "http://Administrator:*****@192.168.1.100:9091/main_v1%2f60/_revs_diff" failed due to error {code,503}
[couchdb:info,2015-02-12T19:24:58.032,ns_1@couchbase.dev.bestforme.com:<0.10497.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f60/_revs_diff in 0.25 seconds due to error {code,503}
[couchdb:info,2015-02-12T19:24:58.157,ns_1@couchbase.dev.bestforme.com:<0.10479.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f18/_bulk_docs in 0.5 seconds due to error {code,503}
[couchdb:info,2015-02-12T19:24:58.285,ns_1@couchbase.dev.bestforme.com:<0.10497.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f60/_revs_diff in 0.5 seconds due to error {code,503}
```


debug.log prints the following:
```
[ns_server:debug,2015-02-12T20:24:01.961,ns_1@couchbase.dev.bestforme.com:<0.5409.14>:remote_clusters_info:do_mk_json_get:1527]Doing get of http://192.168.1.100:9091/pools/default/buckets?uuid=dc197740d72e32c7abc6b40b0890baf4&forXDCR=1
[ns_server:debug,2015-02-12T20:24:01.963,ns_1@couchbase.dev.bestforme.com:<0.5409.14>:remote_clusters_info:do_with_mcd_to_couch_uri_dict:1414]Going to pick plain couchApiBase for mcd_to_couch_uri_dict: <<"http://192.168.1.100:9091/main_v1">>
[ns_server:debug,2015-02-12T20:24:01.966,ns_1@couchbase.dev.bestforme.com:<0.5408.14>:xdc_vbucket_rep:init_replication_state:650]Inited replication position: {0,0,0,0,<<"be9367b897fb4e38aa7884e99ccf27f7">>}
[ns_server:debug,2015-02-12T20:24:02.728,ns_1@couchbase.dev.bestforme.com:remote_clusters_info<0.685.0>:remote_clusters_info:handle_call:369]FoundCluster: {remote_cluster,<<"dc197740d72e32c7abc6b40b0890baf4">>,
                  [{remote_node,"192.168.1.100",9091,9091,undefined,
                       undefined,undefined}],
                  undefined}
[ns_server:debug,2015-02-12T20:24:02.728,ns_1@couchbase.dev.bestforme.com:<0.5430.14>:remote_clusters_info:do_mk_json_get:1527]Doing get of http://192.168.1.100:9091/pools
[error_logger:error,2015-02-12T20:24:02.728,ns_1@couchbase.dev.bestforme.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]** Generic server <0.5408.14> terminating 
** Last message in was {'EXIT',<0.5418.14>,
                           {http_request_failed,"POST",
                               "http://Administrator:*****@192.168.1.100:9091/main_v1%2f63/_bulk_docs",
                               {error,{code,503}}}}
** When Server state == [{data,
                          [{"State",
                            {rep_state,
                             {rep,
                              <<"dc197740d72e32c7abc6b40b0890baf4/main/main_v1">>,
                              <<"main">>,
                              <<"/remoteClusters/dc197740d72e32c7abc6b40b0890baf4/buckets/main_v1">>,
                              "capi",
                              [{optimistic_replication_threshold,256},
                               {worker_processes,4},
                               {worker_batch_size,500},
                               {failure_restart_interval,30},
                               {doc_batch_size_kb,2048},
                               {checkpoint_interval,1800},
                               {max_concurrent_reps,16},
                               {connection_timeout,180},
                               {http_connections,20},
                               {retries_per_request,2},
                               {socket_options,
                                [{keepalive,true},{nodelay,false}]},
                               {pause_requested,false},
                               {supervisor_max_r,25},
                               {supervisor_max_t,5}]},
                             {rep_vb_status,63,<0.5408.14>,replicating},
                             {1423,790641,966684},
                             nil,nil,<0.11052.10>,<0.11049.10>,<<"main/63">>,
                             <<"*****@192.168.1.100:9091/main_v1%2f63">>,
                             {httpdb,
                              "*****@192.168.1.100:9091/main_v1%2f63/",nil,
                              [{"Accept","application/json"},
                               {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                              180000,
                              [{connect_options,
                                [{keepalive,true},{nodelay,false}]},
                               {connect_timeout,180000},
                               {send_retry,0}],
                              2,250,xdc_lhttpc_pool,20},
                             {db,<0.1830.0>,<0.1831.0>,nil,
                              <<"1423773839449534">>,<0.1826.0>,<0.1836.0>,
                              {db_header,11,0,nil,nil,
                               <<0,0,0,0,97,216,0,0,0,0,5,139>>,
                               0,nil,nil},
                              0,
                              {btree,<0.1826.0>,nil,
                               #Fun<couch_db_updater.7.78420415>,
                               #Fun<couch_db_updater.8.78420415>,
                               #Fun<couch_btree.1.39972947>,
                               #Fun<couch_db_updater.9.78420415>,1279,2558,
                               true},
                              {btree,<0.1826.0>,nil,
                               #Fun<couch_db_updater.10.78420415>,
                               #Fun<couch_db_updater.11.78420415>,
                               #Fun<couch_db_updater.6.78420415>,
                               #Fun<couch_db_updater.12.78420415>,1279,2558,
                               true},
                              {btree,<0.1826.0>,
                               {25048,<<>>,1419},
                               identity,identity,#Fun<couch_btree.1.39972947>,
                               nil,1279,2558,true},
                              0,<<"main/master">>,
                              "/Users/Levon/Library/Application Support/Couchbase/var/lib/couchdb/main/master.couch.1",
                              [],nil,
                              {user_ctx,null,[],undefined},
                              nil,
                              [before_header,after_header,on_file_open],
                              [create]},
                             <<"be9367b897fb4e38aa7884e99ccf27f7">>,0,
                             undefined,0,0,4230,4230,0,13797118081439,
                             [4230],
                             0,
                             {0,0},
                             "Fri, 13 Feb 2015 01:24:01 GMT",
                             {1423792441967640,#Ref<0.0.20.120150>},
                             {1423,790641,967308},
                             {1423,790641,966710},
                             [<0.5418.14>],
                             <0.5415.14>,false,
                             {"http://Administrator:kva%24im0d0@192.168.1.100:9091/",
                              [{<<"vb">>,63},
                               {<<"bucket">>,<<"main_v1">>},
                               {<<"bucketUUID">>,
                                <<"fc70a8224cc64abe933b826a29c8bba3">>}],
                              {httpdb,
                               "*****@192.168.1.100:9091/main_v1%2f63/",nil,
                               [{"Accept","application/json"},
                                {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                               180000,
                               [{connect_options,
                                 [{keepalive,true},{nodelay,false}]},
                                {connect_timeout,180000},
                                {send_retry,0}],
                               2,250,xdc_lhttpc_pool,20}}}}]}]
** Reason for termination == 
** {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f63/_bulk_docs",
                        {error,{code,503}}}

[ns_server:debug,2015-02-12T20:24:02.729,ns_1@couchbase.dev.bestforme.com:<0.5430.14>:remote_clusters_info:do_mk_json_get:1527]Doing get of http://192.168.1.100:9091/pools/default?uuid=dc197740d72e32c7abc6b40b0890baf4
[error_logger:error,2015-02-12T20:24:02.732,ns_1@couchbase.dev.bestforme.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]
=========================CRASH REPORT=========================
  crasher:
    initial call: xdc_vbucket_rep:init/1
    pid: <0.5408.14>
    registered_name: []
    exception exit: {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f63/_bulk_docs",
                        {error,{code,503}}}
      in function  gen_server:terminate/6 (gen_server.erl, line 744)
    ancestors: [<0.11053.10>,<0.11049.10>,xdc_replication_sup,xdcr_sup,
                  ns_server_sup,ns_server_cluster_sup,<0.59.0>]
    messages: []
    links: [<0.11053.10>]
    dictionary: [{changes_reader,<0.5416.14>}]
    trap_exit: true
    status: running
    heap_size: 6772
    stack_size: 27
    reductions: 13694
  neighbours:

[error_logger:info,2015-02-12T20:24:02.732,ns_1@couchbase.dev.bestforme.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]
=========================PROGRESS REPORT=========================
          supervisor: {<0.11053.10>,xdc_vbucket_rep_sup}
             started: [{pid,<0.5429.14>},
                       {name,63},
                       {mfa,
                        {xdc_vbucket_rep,start_link,
                         [{rep,
                           <<"dc197740d72e32c7abc6b40b0890baf4/main/main_v1">>,
                           <<"main">>,
                           <<"/remoteClusters/dc197740d72e32c7abc6b40b0890baf4/buckets/main_v1">>,
                           "capi",
                           [{optimistic_replication_threshold,256},
                            {worker_processes,4},
                            {worker_batch_size,500},
                            {failure_restart_interval,30},
                            {doc_batch_size_kb,2048},
                            {checkpoint_interval,1800},
                            {max_concurrent_reps,16},
                            {connection_timeout,180},
                            {http_connections,20},
                            {retries_per_request,2},
                            {socket_options,
                             [{keepalive,true},{nodelay,false}]},
                            {pause_requested,false},
                            {supervisor_max_r,25},
                            {supervisor_max_t,5}]},
                          63,<0.11051.10>,<0.11052.10>,<0.11049.10>,"capi"]}},
                       {restart_type,{permanent,30}},
                       {shutdown,100},
                       {child_type,worker}]

[ns_server:debug,2015-02-12T20:24:02.732,ns_1@couchbase.dev.bestforme.com:<0.5430.14>:remote_clusters_info:do_mk_json_get:1527]Doing get of http://192.168.1.100:9091/pools/default/buckets?uuid=dc197740d72e32c7abc6b40b0890baf4&forXDCR=1
[ns_server:debug,2015-02-12T20:24:02.735,ns_1@couchbase.dev.bestforme.com:<0.5430.14>:remote_clusters_info:do_with_mcd_to_couch_uri_dict:1414]Going to pick plain couchApiBase for mcd_to_couch_uri_dict: <<"http://192.168.1.100:9091/main_v1">>
[ns_server:debug,2015-02-12T20:24:02.737,ns_1@couchbase.dev.bestforme.com:<0.5429.14>:xdc_vbucket_rep:init_replication_state:650]Inited replication position: {0,0,0,0,<<"be9367b897fb4e38aa7884e99ccf27f7">>}
 ```

Any guidance would be really appreciated. I could send over the full logs if that helps.

Thank you,

**Comment by Branor** [comment]
Fixed in v2.1.1
